File: rb/features.rb
# VGd SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module VGdFeatures
  def self.make_feature(name)
    case name
    when "base"
      VGdBaseFeature.new
    when "test"
      VGdTestFeature.new
    else
      VGdBaseFeature.new
    end
  end
end

File: py/features.py
# VGd SDK feature factory

from feature.base_feature import VGdBaseFeature
from feature.test_feature import VGdTestFeature


def _make_feature(name):
    features = {
        "base": lambda: VGdBaseFeature(),
        "test": lambda: VGdTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
